Agents/CEAAgent/README.md

```diff
@@ -0,0 +1,386 @@
+For building height, there are three different possible queries that can retrieve the building height. Each of the following queries are tried, in the order they are listed, until a result is retrieved. If all the queries return empty results, a default value of 10.0m for building height is set.
+
+First query to for building height:
+```
+PREFIX  ocgml: <http://www.theworldavatar.com/ontology/ontocitygml/citieskg/OntoCityGML.owl#>
+
+SELECT  ?Height
+WHERE
+  { GRAPH <{PREFIX}building/>
+      { <{PREFIX}building/{UUID}/> ocgml:measuredHeigh  ?Height}}
+```
+
+Second query to try for building height:
+```
+PREFIX  ocgml: <http://www.theworldavatar.com/ontology/ontocitygml/citieskg/OntoCityGML.owl#>
+
+SELECT  ?Height
+WHERE
+  { GRAPH <{PREFIX}building/>
+      { <{PREFIX}building/{UUID}/> ocgml:measuredHeight  ?Height}}
+```
+
+Third query to try for building height:
+```
+PREFIX  ocgml: <http://www.theworldavatar.com/ontology/ontocitygml/citieskg/OntoCityGML.owl#>
+
+SELECT  ?Height
+WHERE
+  { GRAPH <{PREFIX}cityobjectgenericattrib/>
+      { ?o  ocgml:attrName      "height" ;
+            ocgml:realVal       ?Height ;
+            ocgml:cityObjectId  <{PREFIX}cityobject/{UUID}/>}}
+
+```
+
+### Building Usage
+The agent queries for the building usage, and the usage share if the building is a multi-usage building, with the following query:
+```
+PREFIX ontobuiltenv: <https://www.theworldavatar.com/kg/ontobuiltenv/>
+
+SELECT  ?BuildingUsage ?UsageShare
+WHERE
+  { ?building  ontobuiltenv:hasOntoCityGMLRepresentation  <{PREFIX}building/{UUID}/> ;
+              ontobuiltenv:hasPropertyUsage  ?usage .
+    ?usage a ?BuildingUsage
+    OPTIONAL
+      { ?usage ontobuiltenv:hasUsageShare ?UsageShare}
+  }
+ORDER BY DESC(?UsageShare)
+```
+
+If no building usage is returned from the query, the default value of ```MULTI_RES``` building usage is set, consistent with the default building usage type used by the CEA. In the case of multiple usages for one building, the OntoBuiltEnv usage concepts are first mapped to the CEA defined usage types according to the mapping at the bottom section of this README; then, since CEA only allows up to three usage types per building, the top three usages and their usage share are passed to CEA as input.
+
+
+### Visualisation
+The 3dWebMapClient can be set up to visualise data produced by the CEA Agent (instructions to run are [here](https://github.com/cambridge-cares/CitiesKG/tree/develop/agents#3dcitydb-web-map-client)). The City Information Agent (CIA) is used when a building on the 3dWebMapClient is selected, to query data stored in the KG on the building. If the parameter "context=energy" is included in the url, the query endpoint of CEA will be contacted for energy data. eg `http://localhost:8000/3dwebclient/index.html?city=kingslynn&context=energy` (NB. this currently requires running web map client and CitiesKG/agents from [develop branch](https://github.com/cambridge-cares/CitiesKG/tree/develop/agents))
```

Comment:
This is clearly something we need to work on, but not as part of this pull request...